### PR TITLE
Add testing for EnableFieldTrie in state_trie.go

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Test addition

**What does this PR do? Why is it needed?**
This test covers tests in state_trie.go for the case that `EnableStateTrie` is enabled. Also adds extra coverage to the other functions.

### CODE REVIEW

File: beacon-chain/state/state_trie_test.go [hunk 1]
```diff
@@ -1,12 +1,21 @@
-package state
+package state_test
 
 import (
+	"bytes"
+	"context"
 	"testing"
 
+	"github.com/prysmaticlabs/prysm/shared/params"
+
```
Comment (terencechain) [inline beacon-chain/state/state_trie_test.go:9]: goimport

Comment (0xKiwi) [reply beacon-chain/state/state_trie_test.go]: Fixed, gotta fix goland setup lol

File: beacon-chain/state/state_trie_test.go [hunk 2]
```diff
@@ -83,3 +103,138 @@ func TestInitializeFromProtoUnsafe(t *testing.T) {
...
+				return beaconState, nil
+			},
+			error: "",
+		},
+	}
+
+	var err error
+	var oldHTR []byte
+	for _, tt := range initTests {
+		t.Run(tt.name, func(t *testing.T) {
```
Comment (nisdas) [inline beacon-chain/state/state_trie_test.go:153]: One more good thing would be to add would be that the generic ssz hash tree root method also gives the same result.

File: beacon-chain/state/state_trie_test.go [hunk 3]
```diff
@@ -83,3 +103,138 @@ func TestInitializeFromProtoUnsafe(t *testing.T) {
...
+				t.Errorf("Unexpected error, expected %v, recevied %v", tt.error, err)
+			}
+			root, err := testState.HashTreeRoot(context.Background())
+			if err == nil && tt.error != "" {
+				t.Errorf("Expected error, expected %v, recevied %v", tt.error, err)
+			}
+			if bytes.Equal(root[:], []byte{}) {
+				t.Error("Received empty hash tree root")
+			}
+			if len(oldHTR) != 0 && bytes.Equal(root[:], oldHTR) {
```
Comment (nisdas) [inline beacon-chain/state/state_trie_test.go:234]: same here, also testing the generic ssz hash tree root would be better.

File: beacon-chain/state/state_trie_test.go (hunk 2 above)
Comment (0xKiwi) [reply beacon-chain/state/state_trie_test.go:154]: Good idea, added!

**Pull request merged**